src/file-table-engine/src/engine/immutable.rs

```diff
@@ -0,0 +1,308 @@
+
+        let table = Arc::new(
+            ImmutableFileTable::create(
+                &table_name,
+                &table_dir,
+                table_info,
+                self.object_store.clone(),
+            )
+            .await?,
+        );
+
+        logging::info!(
+            "Immutable file engine created table: {} in schema: {}, table_id: {}.",
+            table_name,
+            schema_name,
+            table_id
+        );
+
+        self.tables
+            .write()
+            .unwrap()
+            .insert(table_ref.to_string(), table.clone());
+
+        Ok(table)
+    }
+
+    fn get_table(&self, table_ref: &TableReference) -> Option<TableRef> {
+        self.tables
+            .read()
+            .unwrap()
+            .get(&table_ref.to_string())
+            .cloned()
+            .map(|table| table as _)
+    }
+
+    async fn open_table(
+        &self,
+        _ctx: &EngineContext,
+        request: OpenTableRequest,
+    ) -> TableResult<Option<TableRef>> {
+        let OpenTableRequest {
+            catalog_name,
+            schema_name,
+            table_name,
+            ..
+        } = request;
+        let table_ref = TableReference {
+            catalog: &catalog_name,
+            schema: &schema_name,
+            table: &table_name,
+        };
+
+        if let Some(table) = self.get_table(&table_ref) {
+            return Ok(Some(table));
+        }
+
+        let table = {
+            let _lock = self.table_mutex.lock().await;
+            // Checks again, read lock should be enough since we are guarded by the mutex.
+            if let Some(table) = self.get_table(&table_ref) {
+                return Ok(Some(table));
+            }
+
+            let table_id = request.table_id;
+            let table_dir = table_dir(&catalog_name, &schema_name, table_id);
+
+            let Some((manifest, table_info)) = self
+                .recover_table_manifest_and_info(&table_name, &table_dir)
+                .await.map_err(BoxedError::new)
+                .context(TableOperationSnafu)? else { return Ok(None) };
+
+            debug!(
+                "Opening table {}, table info recovered: {:?}",
+                table_id, table_info
+            );
+
+            let table = Arc::new(ImmutableFileTable::new(table_info, manifest));
+
+            self.tables
+                .write()
+                .unwrap()
+                .insert(table_ref.to_string(), table.clone());
+            Some(table as _)
+        };
+
+        logging::info!(
+            "Mito engine opened table: {} in schema: {}",
+            table_name,
+            schema_name
+        );
+
+        Ok(table)
+    }
+
+    async fn drop_table(&self, req: DropTableRequest) -> Result<bool> {
+        let table_reference = TableReference {
+            catalog: &req.catalog_name,
+            schema: &req.schema_name,
+            table: &req.table_name,
+        };
+        //TODO(weny): reclaim persisted data
```

Comment:
suggestion: We could fix this TODO in the file table engine. It shouldn't be too difficult to drop this table physically, especially if we simplified the implementation of the manifest file. We just need to remove that file and the table dir.